src/net/dmytro/homework4/Cat.java

```diff
@@ -1,31 +1,36 @@
 package net.dmytro.homework4;
 
 public class Cat extends Animal {
-private String catName;
-private static int count = 0;
+private String name;
+private static int count;
+
+    public Cat(String name) {
+        this.name=name;
+        count++;
+
 
-    public Cat(String catName) {
-        this.catName = catName;
     }
 
+
     @Override
-    void ran(int maxRan) {
-        if (maxRan >= 0 ) {
-            System.out.println("Cat " + catName +" run " + maxRan + " м");
+    void ran(int maxRanDistance, int distanceToDo) {
+        if (maxRanDistance >= distanceToDo && distanceToDo >0 ) {
+            System.out.println("Cat " + name +" run " + distanceToDo + " м");
         }
-        count++;
+
```

Comment:
here you can add else block, in case it cannot run(or swim in another method ) so much